### PR TITLE
[improvement] Feature/debug logs

### LINKED ISSUE #238: [Feature] Debug settings UI
### Describe the problem

For the new Settings UI, we need a "Debug" section.

### Describe the proposed change

A page in the new Settings UI that has:
- A live log feed (from the static OwlCore.Services.Logger)
- A button to open the logs folder
- A button to soft reset the app (delete all user data)
- Show the current window size (will be used to debug UI issues reported by users)
- Show memory usage

### Alternatives

n/a

### Additional info

We had something like this in v1, but we need something with a new design
image

If you go to https://toolkitlabs.dev/ you'll find a couple of Settings controls that we use now in v2 that match the Windows 11 style:
image

### Help us help you

None

### PR DESCRIPTION
**Pull request by amaid**
## Overview

Added debug settings and logs in the settings window.
Closes #238

Implemented the following under debug settings:
1. A live log feed (from the static 
2. A button to open the logs folder
3. A button to soft reset the app (delete all user data)
4. Show the current window size (will be used to debug UI issues reported by users)
5. Show memory usage

https://user-images.githubusercontent.com/15306909/209475368-5a17e394-a19d-48fb-b1c1-6a66a5ee043d.mp4

## Checklist

This PR meets the following requirements:
- [x] Tested and contains **NO** breaking changes or known regressions.
- [x] Tested with the upstream branch merged in.
- [x] Tests have been added for bug fixes / features (or this option is not applicable)
- [x] All new code has been documented (or this option is not applicable)
- [x] Headers have been added to all new source files (or this option is not applicable)

## Additional info

Not provided

### CODE REVIEW

File: src/Shared/AppModels/AppDebug.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = $"App Memory used: {SizeSuffix((long)memoryUsed)}, Expected app memory usage limit: {SizeSuffix((long)totalMemory)}.";
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowSizeStr = $"Current size is: Height: {Window.Current.Bounds.Height}, Width: {Window.Current.Bounds.Width}";
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:76]: Let's switch to separate heigh / width properties that we can bind to

File: src/Shared/AppModels/AppDebug.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = $"App Memory used: {SizeSuffix((long)memoryUsed)}, Expected app memory usage limit: {SizeSuffix((long)totalMemory)}.";
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:69]: Let's create separate `MemoryUsed` and `TotalMemory` properties instead of using a string

File: src/Shared/AppModels/AppDebug.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            () =>
+            {
+                AppLogs.Add(formatedMessage);
+            });
+
+            _semaphoreSlim.Release();
+        }
+
+        [RelayCommand]
+        private async void OpenLogFolder()
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:98]: Don't use `async void` unless you're writing an event handler (this is not an event handler, this is a normal method).

Use `async Task` instead, and `[RelayCommand]` will generate an `AsyncRelayCommand` you can bind to.

File: src/Shared/AppModels/AppDebug.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        {
+            var logsLocation = LogFormatter.LogFolderPath;
+            var folder = await StorageFolder.GetFolderFromPathAsync(logsLocation);
+
+            await Launcher.LaunchFolderAsync(folder);
+        }
+
+        [RelayCommand]
+        private async void DeleteUserData()
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:107]: Don't use `async void` unless you're writing an event handler (this is not an event handler, this is a normal method).

Use `async Task` instead, and `[RelayCommand]` will generate an `AsyncRelayCommand` you can bind to.

File: src/Shared/AppModels/AppDebug.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                Content = "Are you sure you want to delete user data? If you hit yes, the app will restart.",
+                PrimaryButtonText = "Yes",
+                SecondaryButtonText = "No",
+                PrimaryButtonCommand = PerformUserDataDeletionCommand
+            };
+
+            await contentDialog.ShowAsync();
+        }
+
+        private async void PerformUserDataDeletion()
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:121]: Don't use `async void` unless you're writing an event handler (this is not an event handler, this is a normal method).

Use `async Task` instead, and `[RelayCommand]` will generate an `AsyncRelayCommand` you can bind to.

File: src/Shared/AppModels/AppDebug.cs (hunk 4 above)
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:107]: Rename to `DeleteUserDataAsync`

File: src/Shared/AppModels/AppDebug.cs (hunk 5 above)
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:121]: Rename to `PerformUserDataDeletionAsync`

File: src/Shared/AppModels/AppDebug.cs [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                folder = await StorageFolder.GetFolderFromPathAsync(localCache);
+                await folder.DeleteAsync();
+
+                var result = await CoreApplication.RequestRestartAsync("Application Restart Programmatically.");
+
+                if (result == AppRestartFailureReason.NotInForeground ||
+                   result == AppRestartFailureReason.RestartPending ||
+                   result == AppRestartFailureReason.Other)
+                {
+                    var msgBox = new MessageDialog("Restart Failed. Please restart the app manually.", result.ToString());
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:140]: MessageDialog was made obsolete along with Windows 8. Please use `ContentDialog` instead.

File: src/Shared/AppModels/AppDebug.cs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+                    await msgBox.ShowAsync();
+                }
+            }
+            catch (Exception)
+            {
+                var msgBox = new MessageDialog("Couldn't delete the user data.");
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:146]: MessageDialog was made obsolete along with Windows 8. Please use ContentDialog instead.

Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:146]: Let's log the caught exception, and add it to the `ContentDialog` displayed to the user. Reuse [this](https://github.com/Arlodotexe/strix-music/blob/e3f9f2c001a92f8ddad034cd353c362c71e70cdd/src/Shared/AppModels/AppRoot.cs#L285) code so it matches the layout we have for other error messages. You don't need to add retry functionality.

File: src/Shared/AppModels/AppDebug.cs [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+                await msgBox.ShowAsync();
+            }
+        }
+
+        private string SizeSuffix(long value, int decimalPlaces = 1)
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:151]: Code style and formatting for this method is incorrect for this project. Let's do some basic cleanup.

File: src/Shared/AppModels/AppRoot.cs [hunk 9]
```diff
@@ -90,6 +94,11 @@ public async Task InitAsync(CancellationToken cancellationToken = default)
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            if (_appDebug is null)
+            {
+                _appDebug = new AppDebug();
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppRoot.cs:99]: No need to assign this in `InitAsync`. We want debug functionality to work right away when `AppRoot` is constructed, so let's assign it from the field or the constructor, and remove nullability.

**Review by Arlodotexe — CHANGES_REQUESTED** [review]
Code review. Let's address these, then we'll iterate on the style a bit.

**Comment by amaid** [comment]
New UI
image

File: src/Shared/Controls/DebugLogs.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        public DebugLogs()
+        {
+            this.InitializeComponent();
+            Loaded += DebugLogs_Loaded;
+        }
+
+        /// <summary>
+        /// The data root for this app instance.
+        /// </summary>
+        public AppRoot? AppRoot
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml.cs:27]: Let's pass `AppDebug` instead of the entire `AppRoot`.

File: src/Shared/Controls/DebugLogs.xaml [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+                </core:EventTriggerBehavior>
+            </interactivity:Interaction.Behaviors>
+        </labs:SettingsCard>
+
+        <TextBlock
+			Margin="1,28,0,4"
+			Style="{StaticResource BodyStrongTextBlockStyle}"
+			Text="Logs" />
+
+        <ScrollViewer x:Name="PART_SvLogs"
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:88]: Never put a ScrollViewer around a ListView. ListView has an internal scrollviewer already, and wrapping another one around it forces the ListView to an `auto` height.

At that point, everything is being shown and virtualization does not apply, but the ScrollViewer surrounding the ListView is hiding it.

Please remove this and use the internal ScrollViewer provided by `ListView`.

File: src/Shared/Controls/DebugLogs.xaml [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+                      MaxWidth="1600"
+                      MaxHeight="450" HorizontalScrollBarVisibility="Visible" HorizontalScrollMode="Enabled" >
+            <ListView  ItemsSource="{x:Bind AppRoot.AppDebug.AppLogs, Mode=OneWay}" >
+
+                <ListView.ItemsPanel>
+                    <ItemsPanelTemplate>
+                        <VirtualizingStackPanel>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:95]: Let's swap this out for an `ItemsStackPanel`:

```
     <ListView.ItemsPanel>
         <ItemsPanelTemplate>
             <ItemsStackPanel ItemsUpdatingScrollMode="KeepLastItemInView" VerticalAlignment="Bottom">
                 <ItemsStackPanel.ChildrenTransitions>
                    <TransitionCollection />
                  </ItemsStackPanel.ChildrenTransitions>
             </ItemsStackPanel>
         </ItemsPanelTemplate>
     </ListView.ItemsPanel>
```

`ItemsStackPanel` is the default used by ListView and supports virtualization, but the code above also enables a "grow list from bottom" behavior, demonstrated by the WinUI 2 Gallery app:

image

File: src/Shared/Controls/DebugLogs.xaml [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+                            </VirtualizingStackPanel.ChildrenTransitions>
+                        </VirtualizingStackPanel>
+                    </ItemsPanelTemplate>
+                </ListView.ItemsPanel>
+
+                <ListView.ItemTemplate>
+                    <DataTemplate>
+                        <StackPanel >
+                            <TextBlock Text="{Binding }" TextWrapping="WrapWholeWords"/>
+                            <TextBlock Text="-----------------------------------------------------------------------------------------------------------------------------------------------"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:107]: Please don't use text to emulate UI elements.

We also shouldn't need a separator for every log message. If the log is hard to read, we should tweak the style to compensate.

```suggestion
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+                                <TransitionCollection/>
+                            </VirtualizingStackPanel.ChildrenTransitions>
+                        </VirtualizingStackPanel>
+                    </ItemsPanelTemplate>
+                </ListView.ItemsPanel>
+
+                <ListView.ItemTemplate>
+                    <DataTemplate>
+                        <StackPanel >
+                            <TextBlock Text="{Binding }" TextWrapping="WrapWholeWords"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:106]:
```suggestion
                            <TextBlock Text="{Binding}" TextWrapping="WrapWholeWords"/>
```

File: src/Shared/Controls/DebugLogs.xaml.cs [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            set => SetValue(AppRootProperty, value);
+        }
+
+        /// <summary>
+        /// The backing dependency property for <see cref="AppRoot"/>.
+        /// </summary>
+        public static readonly DependencyProperty AppRootProperty =
+            DependencyProperty.Register(nameof(AppRoot), typeof(AppRoot), typeof(DebugLogs), new PropertyMetadata(null));
+
+        private void DebugLogs_Loaded(object sender, RoutedEventArgs e)
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml.cs:39]: All of this can be removed when you implement https://github.com/Arlodotexe/strix-music/pull/240/files#r1057870486

File: src/Shared/Controls/DebugLogs.xaml.cs [hunk 16]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using StrixMusic.AppModels;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml.cs:6]: Remember to remove the default file placeholders :)

File: src/Shared/Controls/DebugLogs.xaml [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+                </TextBlock>
+            </labs:SettingsCard.Content>
+        </labs:SettingsCard>
+
+        <labs:SettingsCard
+			ActionIcon="{ui:SymbolIcon Symbol=Clear}"
+			Header="Delete User Data"
+			HeaderIcon="{ui:SymbolIcon Symbol=Delete}"
+			IsClickEnabled="True">
+            <interactivity:Interaction.Behaviors>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:51]: Let's use a button instead of using the entire card.

File: src/Shared/Controls/DebugLogs.xaml [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+			Text="Debug Info" />
+
+        <labs:SettingsCard
+			Header="Window Size"
+			IsClickEnabled="True" >
+            <labs:SettingsCard.Content>
+                <TextBlock >
+                    <Run Text="Height(px):"/>
+                    <Run Text="{x:Bind AppRoot.AppDebug.WindowHeight}"/>
+                    <Run Text=" Width(px):"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:40]:
```suggestion
                    <Run Text="{x:Bind AppRoot.AppDebug.WindowWidth, Mode=OneWay}"/>
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+			Style="{StaticResource BodyStrongTextBlockStyle}"
+			Text="Debug Info" />
+
+        <labs:SettingsCard
+			Header="Window Size"
+			IsClickEnabled="True" >
+            <labs:SettingsCard.Content>
+                <TextBlock >
+                    <Run Text="Height(px):"/>
+                    <Run Text="{x:Bind AppRoot.AppDebug.WindowHeight}"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:39]:
```suggestion
                    <Run Text="h x"/>
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+                    <Run Text=" Width(px):"/>
+                    <Run Text="{x:Bind AppRoot.AppDebug.WindowWidth}"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:41]:
```suggestion
                    <Run Text="w"/>
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+			Margin="1,28,0,4"
+			Style="{StaticResource BodyStrongTextBlockStyle}"
+			Text="Debug Info" />
+
+        <labs:SettingsCard
+			Header="Window Size"
+			IsClickEnabled="True" >
+            <labs:SettingsCard.Content>
+                <TextBlock >
+                    <Run Text="Height(px):"/>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:38]:
```suggestion
                    <Run Text="{x:Bind AppRoot.AppDebug.WindowHeight, Mode=OneWay}"/>
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+                    <core:InvokeCommandAction Command="{x:Bind AppRoot.AppDebug.DeleteUserDataCommand, Mode=OneWay}"/>
+                </core:EventTriggerBehavior>
+            </interactivity:Interaction.Behaviors>
+        </labs:SettingsCard>
+
+        <labs:SettingsCard
+			Header="Memory"
+			IsClickEnabled="True" >
+            <labs:SettingsCard.Content>
+                <TextBlock >
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:62]: Try this:
- Change Header to "Memory Usage"
- Put the memory limit in the SettingsCard description
- Remove the remaining text in the content. We only need the MemoryUsage binding.

File: src/Shared/Controls/DebugLogs.xaml [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+                    <Run Text="Memory Used:"/>
+                    <Run Text="{x:Bind AppRoot.AppDebug.MemoryUsage}"/>
+                    <Run Text="Expected Memory Limit:"/>
+                    <Run Text="{x:Bind AppRoot.AppDebug.TotalMemory}"/>
+                </TextBlock>
+            </labs:SettingsCard.Content>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:68]: Accessing the "Content" property is implicit in XAML. You can remove this.
```suggestion
```

Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:68]: Accessing the "Content" property is implicit in XAML. You can remove this.
```suggestion
```

File: src/Shared/Controls/DebugLogs.xaml [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+                <core:EventTriggerBehavior EventName="Click">
+                    <core:InvokeCommandAction Command="{x:Bind AppRoot.AppDebug.DeleteUserDataCommand, Mode=OneWay}"/>
+                </core:EventTriggerBehavior>
+            </interactivity:Interaction.Behaviors>
+        </labs:SettingsCard>
+
+        <labs:SettingsCard
+			Header="Memory"
+			IsClickEnabled="True" >
+            <labs:SettingsCard.Content>
```
Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:61]: Accessing the "Content" property is implicit in XAML. You can remove this.
```suggestion
```

Comment (Arlodotexe) [inline src/Shared/Controls/DebugLogs.xaml:61]: Accessing the "Content" property is implicit in XAML. You can remove this.
```suggestion
```

**Review by Arlodotexe — CHANGES_REQUESTED** [review]

File: src/Shared/AppModels/AppDebug.cs [hunk 25]
```diff
@@ -0,0 +1,229 @@
...
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = SizeSuffix((long)memoryUsed);
+            TotalMemory = SizeSuffix((long)totalMemory);
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowHeight = Window.Current.Bounds.Height.ToString(); ;
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:100]:
```suggestion
            WindowHeight = Window.Current.Bounds.Height.ToString();
```

File: src/Shared/AppModels/AppDebug.cs [hunk 26]
```diff
@@ -0,0 +1,229 @@
...
+            WindowWidth = Window.Current.Bounds.Width.ToString();
+        }
+
+        private async void Logger_MessageReceived(object sender, LoggerMessageEventArgs e)
+        {
+            await _semaphoreSlim.WaitAsync();
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:106]: Let's replace this with the `.DisposableWaitAsync()` extension method in OwlCore.Extensions. You forgot to release the semaphore before returning a few lines down.

Doing `using (await _semaphore.DisposableWaitAsync() { ... }` will remove the need to explicitly release the semaphore.

File: src/Shared/AppModels/AppDebug.cs [hunk 27]
```diff
@@ -0,0 +1,229 @@
...
+            _deleteUserDataDialog = new ContentDialog()
+            {
+                Title = "Delete user data",
+                Content = "Are you sure you want to delete user data? If you hit yes, the app will restart.",
+                PrimaryButtonText = "Yes",
+                SecondaryButtonText = "No",
+                PrimaryButtonCommand = PerformUserDataDeletionCommand
+            };
+
+            await _deleteUserDataDialog.ShowAsync();
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:143]: Please merge `feature/settings-ui` into this branch, and add a `ShowType` enum in the first parameter. This new extension method allows us to queue or interrupt any `ContentDialog` that's already open, instead of the default behavior of just crashing.

File: src/Shared/AppModels/AppDebug.cs [hunk 28]
```diff
@@ -0,0 +1,229 @@
...
+                                Content = new TextBlock { Text = $"Reason: {ex}", FontSize = 11, IsTextSelectionEnabled = true },
+                            },
+                        },
+                },
+                CloseButtonText = "Ignore, remove source",
+                PrimaryButtonText = "Try again",
+                PrimaryButtonCommand = retryCommand,
+            };
+
+            await retryConfirmationDialog.ShowAsync();
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:205]: Please merge feature/settings-ui into this branch, and add a ShowType enum in the first parameter. This new extension method allows us to queue or interrupt any ContentDialog that's already open, instead of the default behavior of just crashing.

File: src/Shared/AppModels/AppDebug.cs [hunk 29]
```diff
@@ -0,0 +1,229 @@
...
+
+            await _deleteUserDataDialog.ShowAsync();
+        }
+
+        [RelayCommand]
+        private async Task PerformUserDataDeletionAsync()
+        {
+            try
+            {
+                var localDataFolder = ApplicationData.Current.LocalFolder.Path;
```
Comment (Arlodotexe) [inline src/Shared/AppModels/AppDebug.cs:151]: The folder being emptied needs to populated from a property on `AppDebug`, and it should be assigned the `IModifiableFolder` passed into the constructor of `AppRoot`.

In OwlCore.Storage, a folder can only be deleted if we know what to delete it from. In other words, we need a reference to the parent folder. Some folder implementations aren't addressable and have no parent, so this won't work. 

Instead, since we're given an `IModifiableFolder`, we can instead delete the _contents_ of the folder, rather than the folder itself.

**Review by Arlodotexe — CHANGES_REQUESTED** [review]

File: src/Shared/AppModels/AppDebug.cs (hunk 27 above)
Comment (amaid) [reply src/Shared/AppModels/AppDebug.cs:143]: Upon testing: QueueNext didn't work as expected 
image
image

ShowType Inturrupted works but crashes on retry by saying _inturrupted dialog is null 
`Parameter "_interruptedDialog" (Windows.UI.Xaml.Controls.ContentDialog) must be null, was <Windows.UI.Xaml.Controls.ContentDialog> (Windows.UI.Xaml.Controls.ContentDialog).`
Parameter name: _interruptedDialog
in short both of them have issues
i'll undo my changes for now and let you fix these first

File: src/Shared/AppModels/AppDebug.cs (hunk 28 above)
Comment (amaid) [reply src/Shared/AppModels/AppDebug.cs:205]: Upon testing: QueueNext didn't work as expected
[image](https://user-images.githubusercontent.com/15306909/209938519-ba4ac8c6-5d4e-45ce-a83c-d3fcb14a6852.png)
[image](https://user-images.githubusercontent.com/15306909/209938584-4f2dd046-70c4-4936-8476-35954f87d664.png)

ShowType Inturrupted works but crashes on retry by saying _inturrupted dialog is null
Parameter "_interruptedDialog" (Windows.UI.Xaml.Controls.ContentDialog) must be null, was <Windows.UI.Xaml.Controls.ContentDialog> (Windows.UI.Xaml.Controls.ContentDialog).
Parameter name: _interruptedDialog
in short both of them have issues
i'll undo my changes for now and let you fix these first

**Review by Arlodotexe — APPROVED** [review]

**Comment by Arlodotexe** [comment]
Added some finishing touching. Renames, style updates, general cleanup. Great work @amaid, you've given us a big boost towards finishing the new Settings UI!
